### ISSUE TITLE
Bug - Sev 3 -  Unknown - User Report - Secure messaging, MHV login and Biometrics

### ISSUE THREAD

**Issue by timwright12** [department-of-veterans-affairs/va-mobile-app#5986]
Whomever picks this up needs to reach out to Adam prior to starting it 

**From Louis Ruiz:**

A Veteran comes in and is using the Health and Benefits app but can't send a SM because he needs a Premium account. Louis verified that he indeed has a Premium account ( Manage MHV Users). Veteran logs into the MHV web site and he has a SM and is able to send/receive SMs in the traditional way through MHV.

The Veteran logs out and log back in to the VA Health and Benefits App and it states he still can't use SM because he needs a premium account!

Louis saw that he used a TOUCH ID to log in to the VA Health and Benefits App and it did not require him to put in his UserID/Password.

Patient went into the settings of the APP, turned off the touch ID and logged out.

This forced him to log in with his UserID/PW for MHV and this gave him the capability to send/receive a Secure Message in the VA Health and Benefits app.

Is it correct to assume that the touch ID/face ID can interfere with a Veterans status IF the Veteran never signed in with their MHV login credentials?

**From Rachel Han:**

Whoever picks up this bug can reach out to Louis Ruiz louis.ruiz@va.gov for additional info if needed, e.g. device, os, version, etc. I learned that he is a VA employee and originator of feedback below

**Discovery done by Tim Wright**

It looks like, from an app standpoint all we do is check the user endpoint and see if the logged in user has "secureMessaging" listed as an authorized service, so the issue would be upstream to us somewhere. I think it's possible that the service that user has hooked up to biometrics isn't the service that gives them access to secure messaging.

**Comment by jperk51** [comment]
Cause of the issue: User needs to have an MHV premium account denoted by `mhv_account_type` on the user object in vets-api. This value is only updated when a user signs in with MHV since the MHV BE system is what upgrades the users account. Once they have done so IDME and logingov accounts will show premium. 

Scenario the user experienced above was likely as follows: Sign in with IDME and set up biometrics. Later set up a premium MHV account. Looked back at the app (IDME sign in via biometrics mhv_account_type != premium). Signed in with MHV on app. Account got upgraded so from then on any sign in method would show mhv_account_type == premium and they would have access to SM. 

@rachelhanster 2 ways to go about solving this issue:
1. Catch this situation with an error message that tells users to sign in with MHV so their account get upgraded. 
2. Work with MHV to expose account upgrade endpoint so SIS can upgrade users as necessary

1 is a much easier solution, but 2 is probably the right solution and would also preemptively take care of something that likely needs to be done anyway for sunsetting MHV as a sign in method. IMO we should implement 1 asap to help users out but also start the ball rolling on 2. 2 requires coordination with MHV and SIS since they are the ones that would be doing the majority, if not all, of the work

**Comment by bischoffa** [comment]
Provided update during Weekly MHV per Rachel's request

**Comment by bischoffa** [comment]
@jperk51 I see this is in Code Review - was there any code chagnes for this ?

**Comment by jperk51** [comment]
I just moved it into "code review" to represent that it was "ready for review"

**Comment by rachelhanster** [comment]
@jperk51 is this problem specific to secure message or all MHV premium related products?

**Comment by jperk51** [comment]
1. I would expect anything that that required MHV premium to be affected
2. According to Trevor, in the thread linked above, handles this situation by showing an error message that informs users to contact an MHV rep and have them manually link their accounts. Specific message of Trevor's: https://dsva.slack.com/archives/CSFV4QTKN/p1689961123683169?thread_ts=1689608324.102329&cid=CSFV4QTKN

**Comment by jperk51** [comment]
One thing I'm going to do is add logging to our SM policy to gather metrics on how often people with IDME/Login.gov accounts aren't showing up as MHV premium. This should help us understand the true scale of the issue.

**Comment by bischoffa** [comment]
@jperk51 can we do that work in this ticket? I can spin off another ticket on other action consideration.

**Comment by jperk51** [comment]
Yeah, it can be part of this ticket. Will only take me a few minutes.

**Comment by rachelhanster** [comment]
@jperk51 you read my mind. That was my next question -- what is the size of this problem?

**Comment by jperk51** [comment]
Now that the logging is in, over the next few days we should start to get a feel for how often this is happening. By monday I should have a significant enough number of records to make an assessment.

**Comment by jperk51** [comment]
I'm putting up a PR to record a couple more fields. Mainly I'm seeing 3 different instances:
1. User doesn't seem to have an mhv_correlation_id. Expected failure. Updating logs to confirm that's the case since the above code just has it and account_type not show up in the logs. My guess is that's because they are null. Which would make this an expected failure
2. User has mhv_correlation_id, but account_type is showing up as "Error". Trevor from SIS is investigating why this might be showing up as an error. Could possibly be the cause of the bug in this ticket.  
3. User has id and account_type is showing up as "Premium". Adding additional logging for the other pieces of info access checks for to see if one of those other fields is causing the issue.

**Comment by jperk51** [comment]
I'll be digging into the logs today and will add my final assessment by EOD.

**Comment by jperk51** [comment]
There are 2 things that could have been causing the above issue:
1. If a user creates an id proofed account but doesn't use it to sign into MHV's site then their account won't be upgraded. [Thread of details](https://dsva.slack.com/archives/C0506QKMW4B/p1691518332202289). This is handled by web by showing users an error message requesting that they call MHV to have support manually upgrade their account to premium and tie it to their existing MHV account. 
2. Logs indicate a small % of people hitting MHV have all the correct attributes but are still getting denied access to SM. I will create a ticket to investigate this. The % is extremely small. 32 total instances in the last week and some are the same user multiple times. 


@rachelhanster I'm planning to close this ticket by EOD tomorrow. If you think I should leave it open let me know,

**Comment by rachelhanster** [comment]
Thanks @jperk51. 
@bischoffa On issue 1, the previously proposed short term and long term solutions make sense. Let's spin up separate tickets please

**Comment by jperk51** [comment]
Closing with understanding that new tickets will be created to move forward.

**Comment by timwright12** [comment]
@jperk51 Can you work with @bischoffa to make sure he has all the info he needs to spin up future issues from this?

**Comment by jperk51** [comment]
All the info needed should be in this ticket, but yeah @bischoffa let me know if you need anything.

**Comment by bischoffa** [comment]
Planning 6497 for next sprint in the bug bucket

**Comment by jperk51** [comment]
> @jperk51 I saw you spun up a ticket for scenario 2. Should we spin up one for scenario 1 as well?

Yup, that's what Rachel suggested above

**Comment by bischoffa** [comment]
Thanks, wanted to make sure one wasnt since you made another ticket for issue 2.

**Comment by timwright12** [comment]
Can we link up that second issue here somewhere for PO tracking? I see 6497, but not sure what the other one is

**Comment by bischoffa** [comment]
@timwright12 Yep just made them. I also updated the ticket description above to call out the tickets. Per Jayson there is no BE on this item only front end.   
  
Scenario 1 If a user creates an id proofed account but doesn't use it to sign into MHV's site then their account won't be upgraded. Thread of details. This is handled by web by showing users an error message requesting that they call MHV to have support manually upgrade their account to premium and tie it to their existing MHV account.

1.  Ticket  [6570](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/6570) - UX Design and Content for Error message
2.  Ticket [6571](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/6571)\- FE Implementation of Error Message

**Comment by timwright12** [comment]
perfect, thank you!

**Comment by rachelhanster** [comment]
@bischoffa can we please track the following with a separate ticket?

Mobile works with MHV to expose account upgrade endpoint, so SIS can upgrade users as necessary. this is the actual fix vs asking a user to do work via an error message

**Comment by bischoffa** [comment]
Created long term ticket [67245](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/6725) 
- SM - Authentication Issues - MHV to expose upgrade endpoint to to SiS
- Will require MHV and SIS to prioritization so the MHV can expose account upgrade endpoint for SIS to use

**Comment by bischoffa** [comment]
Per Jayson, saying this only impacts SM isn't exactly correct. It does apply to all MHV health services, but the ones that show up on MHV and not on mobile for those effected by this would typically only be those features gated by needing to have an MHV "premium" account.
